src/main/java/com/google/sps/servlets/ScheduledInterviewServlet.java

```diff
@@ -17,59 +17,78 @@
 import com.google.appengine.api.users.UserService;
 import com.google.appengine.api.users.UserServiceFactory;
 import com.google.gson.Gson;
+import com.google.sps.data.DatastorePersonDao;
 import com.google.sps.data.DatastoreScheduledInterviewDao;
+import com.google.sps.data.Person;
+import com.google.sps.data.PersonDao;
 import com.google.sps.data.ScheduledInterview;
 import com.google.sps.data.ScheduledInterviewDao;
+import com.google.sps.data.ScheduledInterviewRequest;
 import com.google.sps.data.TimeRange;
 import java.io.IOException;
+import java.time.format.DateTimeFormatter;
+import java.time.format.TextStyle;
 import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Locale;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
 import java.time.format.DateTimeParseException;
 
 @WebServlet("/scheduled-interviews")
 public class ScheduledInterviewServlet extends HttpServlet {
 
   private ScheduledInterviewDao scheduledInterviewDao;
+  private PersonDao personDao;
 
   @Override
   public void init() {
-    init(new DatastoreScheduledInterviewDao());
+    init(new DatastoreScheduledInterviewDao(), new DatastorePersonDao());
   }
 
-  // TODO: add a FakeScheduledInterviewDao class so this will become useful
-  public void init(ScheduledInterviewDao scheduledInterviewDao) {
+  public void init(ScheduledInterviewDao scheduledInterviewDao, PersonDao personDao) {
     this.scheduledInterviewDao = scheduledInterviewDao;
+    this.personDao = personDao;
   }
 
   // Gets the current user's email from request and returns the ScheduledInterviews for that person.
-  // If the email that is requested matches the email that is logged in, then the scheduled
-  // interviews are returned, otherwise SC_UNAUTHORIZED is returned.
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String timeZoneId = request.getParameter("timeZone");
     String userEmail = UserServiceFactory.getUserService().getCurrentUser().getEmail();
     String userId = UserServiceFactory.getUserService().getCurrentUser().getUserId();
+    // Since UserId does not have a valid Mock, if the id is null (as when testing), it will be
```

Comment:
Copying multi line comments multiple times is a red flag. Code duplication is fine sometimes, but if each instance of duplication needs an explanation then that means you ought to be doing something else.

Create a helper somewhere (not here, probably), that gets the user id or a mock id.

Reply:
Disregard, discussed in standup

(this sucks, but I don't think we have time budget to fix it)